oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/web/server/OAuth2AuthorizationCodeGrantWebFilter.java

```diff
@@ -184,6 +186,8 @@ private void updateDefaultAuthenticationConverter() {
 		return this.authenticationManager.authenticate(token)
 				.switchIfEmpty(Mono.defer(() -> Mono.error(new IllegalStateException("No provider found for " + token.getClass()))))
 				.flatMap(authentication -> onAuthenticationSuccess(authentication, webFilterExchange))
+				.onErrorResume(OAuth2AuthorizationException.class, e -> this.authenticationFailureHandler
```

Comment:
This is not the correct place for the fix. `OAuth2AuthorizationException` is not thrown by `AuthenticationManager`, instead it's thrown by `authenticationConverter.convert(exchange)` (`ServerOAuth2AuthorizationCodeAuthenticationTokenConverter`)